### PR TITLE
information-functions: Update reference manual and more

### LINKED ISSUE #16818: Update reference manual links in functions-and-operators/information-functions.md

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

Closes #16818

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: functions-and-operators/information-functions.md [hunk 1]
```diff
@@ -25,6 +26,224 @@ TiDB supports most of the [information functions](https://dev.mysql.com/doc/refm
...
++--------------+
+|            2 |
++--------------+
+1 row in set (0.00 sec)
+```
+
+> **Note**
+>
+> `SQL_CALC_FOUND_ROWS` is only accepted if [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-new-in-v40) is set. This query modifier has been deprecated in MySQL 8.0.17.
+
```
Comment (mjonss) [inline functions-and-operators/information-functions.md:146]: Not sure if there are enough context here?
I assume the SQL_CALC_FOUND_ROWS should have been included in the `SELECT 1 UNION ALL SELECT 2` if there would have been a limit?

File: functions-and-operators/information-functions.md [hunk 2]
```diff
@@ -25,6 +26,224 @@ TiDB supports most of the [information functions](https://dev.mysql.com/doc/refm
...
+### LAST_INSERT_ID()
+
+This returns the ID of the last row that was inserted. This works with [`AUTO_INCREMENT`](/auto-increment.md) and [`AUTO_RANDOM`](/auto-random.md) columns.
+
+```
+mysql> CREATE TABLE t1(id BIGINT UNSIGNED PRIMARY KEY AUTO_RANDOM);
```
Comment (mjonss) [inline functions-and-operators/information-functions.md:152]: Even though this is more interesting, I think an example with AUTO_INCREMENT is more straight forward.

File: functions-and-operators/information-functions.md [hunk 3]
```diff
@@ -25,6 +26,224 @@ TiDB supports most of the [information functions](https://dev.mysql.com/doc/refm
...
+
+mysql> INSERT INTO t1() VALUES (),(),();
+Query OK, 3 rows affected (0.02 sec)
+Records: 3  Duplicates: 0  Warnings: 0
+
+mysql> SELECT LAST_INSERT_ID();
++----------------------+
+| LAST_INSERT_ID()     |
++----------------------+
+| 10952754293765046273 |
```
Comment (mjonss) [inline functions-and-operators/information-functions.md:163]: Now it looks like it have returned the first row of the last INSERT (unless one knows how AUTO_RANDOM generates the numbers).

File: functions-and-operators/information-functions.md [hunk 4]
```diff
@@ -25,6 +26,224 @@ TiDB supports most of the [information functions](https://dev.mysql.com/doc/refm
...
+1 row in set (0.00 sec)
+```
+
+> **Note**
+>
+> `SQL_CALC_FOUND_ROWS` is only accepted if [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-new-in-v40) is set. This query modifier has been deprecated in MySQL 8.0.17.
+
+### LAST_INSERT_ID()
+
+This returns the ID of the last row that was inserted. This works with [`AUTO_INCREMENT`](/auto-increment.md) and [`AUTO_RANDOM`](/auto-random.md) columns.
```
Comment (mjonss) [inline functions-and-operators/information-functions.md:149]: Is TiDB different from MySQL here?
In MySQL it is the first generated ID (from the first statement if multi-statement is sent).

File: functions-and-operators/information-functions.md [hunk 5]
```diff
@@ -25,6 +26,224 @@ TiDB supports most of the [information functions](https://dev.mysql.com/doc/refm
...
+| USER()         | CURRENT_USER() |
++----------------+----------------+
+| root@127.0.0.1 | root@%         |
++----------------+----------------+
+1 row in set (0.00 sec)
+```
+
+### VERSION()
+
+This function returns the function of TiDB in a way that is compatible with MySQL. The [`TIDB_VERSION()`](/functions-and-operators/tidb-functions.md#tidb_version) function can be used to get more details.
```
Comment (mjonss) [inline functions-and-operators/information-functions.md:218]:
```suggestion
This function returns the version of TiDB in a way that is compatible with MySQL. The [`TIDB_VERSION()`](/functions-and-operators/tidb-functions.md#tidb_version) function can be used to get more details.
```

**Review by mjonss — APPROVED** [review]
LGTM, with some questions and suggestions.

**Comment by dveeden** [comment]
@bb7133 Updated this with information from https://github.com/pingcap/tidb/issues/52214 and https://github.com/pingcap/tidb/issues/982

**Review by bb7133 — APPROVED** [review]
LGTM

**Comment by qiancai** [comment]
Hi @dveeden, I've submitted https://github.com/pingcap/docs/pull/16853/commits for description updates. PTAL. Thanks.

File: functions-and-operators/information-functions.md [hunk 6]
```diff
@@ -6,24 +6,279 @@ aliases: ['/docs/dev/functions-and-operators/information-functions/','/docs/dev/
...
+| FOUND_ROWS() |
++--------------+
+|            2 |
++--------------+
+1 row in set (0.00 sec)
+```
+
+> **Note:**
+>
+> The `SQL_CALC_FOUND_ROWS` query modifier, which calculates the total number of rows  in a result set without considering the `LIMIT` clause, is only accepted if [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-new-in-v40) is enabled. This query modifier is deprecated starting from MySQL 8.0.17.  It is recommended to use `COUNT(*)` instead.
```
Comment (qiancai) [inline functions-and-operators/information-functions.md:160]:
```suggestion
> The `SQL_CALC_FOUND_ROWS` query modifier, which calculates the total number of rows in a result set without considering the `LIMIT` clause, is only accepted if [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-new-in-v40) is enabled. This query modifier is deprecated starting from MySQL 8.0.17.  It is recommended to use `COUNT(*)` instead.
```

File: functions-and-operators/information-functions.md [hunk 7]
```diff
@@ -6,24 +6,279 @@ aliases: ['/docs/dev/functions-and-operators/information-functions/','/docs/dev/
...
+
+The `SCHEMA()` function is a synonym for `DATABASE()`.
+
+### SESSION_USER()
+
+The `SESSION_USER()` function is a synonym for `USER()`.
+
+### SYSTEM_USER()
+
+The `SYSTEM_USER()` function is a synonym for `USER()`.
```
Comment (qiancai) [inline functions-and-operators/information-functions.md:231]:
```suggestion
The `SCHEMA()` function is a synonym for [`DATABASE()`](#database).

### SESSION_USER()

The `SESSION_USER()` function is a synonym for [`USER()`](#user).

### SYSTEM_USER()

The `SYSTEM_USER()` function is a synonym for [`USER()`](#user).
```

File: functions-and-operators/information-functions.md [hunk 8]
```diff
@@ -6,24 +6,279 @@ aliases: ['/docs/dev/functions-and-operators/information-functions/','/docs/dev/
...
+| [`CONNECTION_ID()`](#connection_id) | Return the connection ID (thread ID) for the connection  |
+| [`CURRENT_ROLE()`](#current_role) | Return the role that is in use by the connection |
+| [`CURRENT_USER()`, `CURRENT_USER`](#current_user) | Return the authenticated user name and host name |
+| [`DATABASE()`](#database) | Return the default (current) database name  |
+| [`FOUND_ROWS()`](#found_rows) | For a `SELECT` with a `LIMIT` clause, the number of the rows that are returned if there is no `LIMIT` clause |
+| [`LAST_INSERT_ID()`](#last_insert_id) | Return the value of the `AUTOINCREMENT` column for the last `INSERT`   |
+| [`ROW_COUNT()`](#row_count) | The number of rows affected |
+| [`SCHEMA()`](#schema) | Synonym for `DATABASE()`  |
+| [`SESSION_USER()`](#session-user) | Synonym for `USER()`    |
+| [`SYSTEM_USER()`](#system-user) | Synonym for `USER()`   |
```
Comment (qiancai) [inline functions-and-operators/information-functions.md:25]:
```suggestion
| [`SESSION_USER()`](#session_user) | Synonym for `USER()`    |
| [`SYSTEM_USER()`](#system_user) | Synonym for `USER()`   |
```

**Review by qiancai — APPROVED** [review]

File: functions-and-operators/information-functions.md [hunk 9]
```diff
@@ -6,24 +6,305 @@ aliases: ['/docs/dev/functions-and-operators/information-functions/','/docs/dev/
...
+|  3 |
++----+
+2 rows in set (0.00 sec)
+```
+
+> **Note**
+>
+> - In TiDB, [`AUTO_ID_CACHE`](/auto-increment.md#auto_id_cache) might lead to results that differ from those returned by MySQL. This discrepancy arises because TiDB caches IDs on each node, potentially leading to IDs that are out of order or have gaps. If maintaining strict ID ordering is essential for your application, you can enable [MySQL compatible mode](/auto-increment.md#mysql-compatibility-mode).
+>
+> - In the preceding example, IDs increase by 2 while MySQL would generate IDs incrementing by one in the same scenario. For more compatibility information, see [Auto-increment ID](/mysql-compatibility.md#auto-increment-id).
```
Comment (hfxsd) [inline functions-and-operators/information-functions.md:220]:
```suggestion
> - In the preceding example, IDs increase by 2 while MySQL would generate IDs incrementing by 1 in the same scenario. For more compatibility information, see [Auto-increment ID](/mysql-compatibility.md#auto-increment-id).
```

**Pull request merged**